bee-common/bee-common/src/packable.rs

```diff
@@ -26,9 +26,49 @@
     }
 
     /// Reads bytes from the passed reader and unpacks them into an instance.
-    fn unpack<R: Read + ?Sized>(reader: &mut R) -> Result<Self, Self::Error>
+    fn unpack_inner<R: Read + ?Sized, const TRUSTED: bool>(reader: &mut R) -> Result<Self, Self::Error>
```

Comment:
I think instead of "TRUSTED" I'ld rather use "CHECKED", the reason being that the concept of "trust" is broad, vague term while "checked" insinuating nothing except that something is ... well .. checked, whether the source being trustful or not.

Reply:
While essentially you cannot trust any byte (as bit flips can happen anywhere anytime) it still can be reasonable to not check a byte for practicality reasons.

Reply:
Yeah but `TRUSTED` is applied to the Reader, not the bytes. A reader can be trusted (storage) or untrusted (network). But tbh I'll probably reduce it down to only `T` as it's going to be used a lot in implementations and would be unnecessarily repetitive.

Reply:
Actually, I think `CHECK` is a better name than `TRUSTED` and `CHECKED`. Instead of qualifying the source or the bytes, it just says if we should apply the checks or not.

Reply:
is fine with me